src/configuration.ts

```diff
@@ -216,14 +237,14 @@ export class Configuration {
     this._serviceContext.version = isString(version) ? version : undefined;
 
     if (isObject(this._givenConfiguration.serviceContext)) {
-      if (isString(this._givenConfiguration.serviceContext.service)) {
-        this._serviceContext.service = this._givenConfiguration.serviceContext.service;
+      if (isString(this._givenConfiguration.serviceContext!.service)) {
+        this._serviceContext.service = this._givenConfiguration.serviceContext!.service!;
```

Comment:
Same question throughout.  Why can we assume these variables are always defined?

Reply:
I think my response above should cover this code too.